pandas/core/generic.py

```diff
@@ -7810,6 +7810,86 @@ def mask(self, cond, other=np.nan, inplace=False, axis=None, level=None,
         is not realigned. That is, use freq if you would like to extend the
         index when shifting and preserve the original data.
 
+        Examples
+        --------
+        Compute the difference between a column in a dataframe
+        with grouped data, and its shifted version.
+
+        >>> data = pd.DataFrame({'myvalue': [1, 2, 3, 4, 5, 6],
+        ...                      'group': ['A', 'A', 'A', 'B', 'B', 'B']},
+        ...                     index=pd.DatetimeIndex(['2016-06-06',
+        ...                                             '2016-06-08',
+        ...                                             '2016-06-09',
+        ...                                             '2016-06-10',
+        ...                                             '2016-06-12',
+        ...                                             '2016-06-13'],
+        ...                                            name='mydate'))
+
+        >>> data
+                    myvalue group
+        mydate
+        2016-06-06        1     A
+        2016-06-08        2     A
+        2016-06-09        3     A
+        2016-06-10        4     B
+        2016-06-12        5     B
+        2016-06-13        6     B
+
+        For the groups compute the difference between current `myvalue` and
+        `myvalue` shifted forward by 1 day.
+
+        If the dataframe is shifted without passing a freq argument than the
+        values simply move down
+
+        >>> data[data.group=='A'].myvalue.shift(1)
+        mydate
+        2016-06-06    NaN
```

Comment:
Others might have a differing opinion but I find these examples rather confusing as you really need to think through how the data is indexed.

I mentioned it before but I think the cleanest approach would be the `reindex` / `fill` set of operations I posted in the original PR - is there any reason why we can't use that here instead of the UDF? I think it more clearly explains the situation and it will certainly scale better on larger datasets, hence why I'd rather we suggest that type of usage in the documentation.

Reply:
Agreed. Your approach seems more explicit. You have suggested
>>> dt_rng = pd.date_range(data.index.min(), data.index.max())
>>> data = data.reindex(rng)
>>> data['group'] = data['group'].ffill()
>>> data.groupby('group')['myvalue'].transform(lambda x: x-x.shift())

I think I can avoid the UDF by doing

>>> df['myvalue'] - df.groupby('group')['myvalue'].shift(1)

I get the same answer, just wanted to confirm that it's the correct thing to do here?
Thanks.

Reply:
That’s correct